Listener.js

```diff
@@ -66,11 +66,30 @@ class Listener {
 
     console.log(`Listening to subscription: ${subscriptionFullName}`);
 
-    const onMessage = (msg) => {
-      onMessageCallback(msg);
-      msg.ack();
+    const onMessageTryCatch = (msg) => {
+      try {
+        onMessageCallback(msg);
+        msg.ack();
+      } catch (err) {
+        console.error(`Error from callback: ${err}\n`);
+        msg.nack();
+        process.exit(1);
+      }
     };
 
+    const onMessageUserHandling = (msg) => {
+      let callback = onMessageCallback(msg);
```

Comment:
this only works when onMessageCallback is synchronous. The response in this instance isn't really a callback. If 
 onMessageCallback was a standard callback function, I'd have thought the structure of the method would be more like:

```
const onMessageUserHandling = (msg) => {
  onMessageCallback(msg, err => {
    if (err) {
      console.error(`Error from callback: ${err}`);
      msg.nack();
      process.exit(1);
    } else {
      msg.ack();
    }
  });
}
```

Reply:
Hi @jfmesse, I took a look at your proposed solution, but it hasn't clicked for me. Apologies, as I am not an experienced Node developer. However, the callback is intended to be provided by the user. It's what you want to be executed when a message is received. I'm not sure how passing an arrow function as an input n the callback addresses this issue.

Since this is an open repo, if you like you may submit your desired changes as a PR and I will evaluate and test them out. Otherwise I'll take a second pass and see what can be done. Thanks for bearing with us.

Reply:
Hi again @jfmesse I took a second look at this, adding the error handling as a callback the way you suggested. I was confused initially, since it's a callback in a callback. Can you see if this works for you?

Reply:
Hi @zakagan , sorry I missed this. Your changes look good to me. We have a developer working on the ingest component now who will be looking to use. Thank you.